content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,326 @@
+
+As an example, take a MongoDB database with two collections, `User` and `Post`. The data in these collections has the following format, with a `userId` field linking users to posts:
+
+`User` collection:
+
+- `_id` field with a type of `objectId`
+- `email` field with a type of `string`
+
+`Post` collection:
+
+- `_id` field with a type of `objectId`
+- `title` field with a type of `string`
+- `userId` with a type of `objectID`
+
+On introspection with `db pull`, this is pulled in to the Prisma schema file as follows:
+
+```prisma file=prisma/schema.prisma
+model Post {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  title  String
+  userId String @db.ObjectId
+}
+
+model User {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+}
+```
+
+This is missing the relation between the `User` and `Post` models. To fix this, manually add a `user` field to the `Post` model with a `@relation` attribute using `userId` as the `fields` value, linking it to the `User` model, and a `posts` field to the `User` model as the back relation:
+
+```prisma file=prisma/schema.prisma highlight=5;add|11;add
+model Post {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  title  String
+  userId String @db.ObjectId
+  user   User   @relation(fields: [userId], references: [id])
+}
+
+model User {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  posts Post[]
+}
+```
+
+For more information on how to use relations in Prisma, see [our documentation](/concepts/components/prisma-schema/relations).
+
+### How to filter for <inlinecode>null</inlinecode> and undefined fields
+
+To understand how MongoDB distinguishes between `null` and undefined fields, consider the example of a `User` model with an optional `name` field:
+
+```ts
+model User {
+  id    String  @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  name  String?
+}
+```
+
+First, try creating a record with the `name` field explicitly set to `null`. Prisma will return `name: null` as expected:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+const createNull = await prisma.user.create({
+  data: {
+    email: 'user1@prisma.io',
+    name: null,
+  },
+})
+console.log(createNull)
+```
+
+</cmd>
+
+<cmdResult>
+
+```
+{
+  id: '6242c4ae032bc76da250b207',
+  email: 'user1@prisma.io',
+  name: null
+}
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+If you check your MongoDB database directly, you will also see a new record with `name` set to `null`:
+
+```terminal
+_id: 6242c4af032bc76da250b207
+email: "user1@prisma.io"
+name: null
+```
```

Comment:
I don't think this is actually terminal output, but probably part of a Json document that should show what is in the database? (Same is true for the following `terminal` ones)

Reply:
+1, I would probably just show the whole document as JSON:

```
{
  "_id": "..."
  ...
}
```

Reply:
Done, agree this is much better